pkg/kubelet/kubelet_pods.go

```diff
@@ -619,6 +619,8 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod
 		return pod.Spec.ServiceAccountName, nil
 	case "status.podIP":
 		return podIP, nil
+	case "status.hostIP":
+		return kl.nodeIP.String(), nil
```

Comment:
`kl.nodeIP` would be nil if you don't set `--node-ip` flag for kubelet. In this case, kubelet will [setNodeAddress](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_node_status.go#L440-L443) by other ways. Can we get `hostIP` from `kl.nodeIP` then?